ee/control/control.go

```diff
@@ -42,6 +42,11 @@ type dataProvider interface {
 	Get(hash string) (data io.Reader, err error)
 }
 
+// StoredDataProvider is an interface for querying data from a persistable data storage layer
+type StoredDataProvider interface {
```

Comment:
I couldn't think of a better name or a better spot for this - it's meant to be a generic enough interface that any places using `*bbolt.DB` references for simple data queries could be replaced with this. `kit` repo is probably a better place, but I didn't want to expand the changes of an already large PR to start touching another repo.

Reply:
Eventually, I'd put this stuff into `pkg/agent/...` but here seems fine for now

Reply:
(well, the interface is used by the consumer, so here seems most correct for that. But if we wanted to think about abstractions, I'd think about the pkg/agent space. TBD!)